### PR TITLE
Move stackdriver debugger to own module

### PR DESCRIPTION
**Pull request by patflynn**
More incremental refactoring work to support #1896.

I had to do this before App Engine as App Engine depends on this and we can't have any extracted module depend on the uber-plugin.

### CODE REVIEW

**Comment by patflynn** [comment]
gradle is not my friend at the moment. my local ./gradlew clean check are consistently green, but every CI run comes up with new failures.

**Comment by patflynn** [comment]
@etanshaul @ivanporty all tests green. this is good to go.

File: google-cloud-tools-plugin/resources/messages/CloudToolsBundle.properties [hunk 1]
```diff
@@ -468,3 +389,5 @@ cloudsdk.check.for.updates.action=Check for updates now
 cloudsdk.location.browse.window.title=Browse to Cloud SDK Directory
 cloudsdk.location.missing.message=Cloud SDK home directory is not specified.
 cloudsdk.download.message={0}Click here{1} to download the Cloud SDK.
+
+clouddebug.logout.additional.message=Any Stackdriver Debug sessions listening in the background will be stopped.
```
Comment (etanshaul) [inline google-cloud-tools-plugin/resources/messages/CloudToolsBundle.properties]: is this is a mistake? (I assume you meant to remove it)

File: google-cloud-tools-plugin/stackdriver-debugger/src/com/google/cloud/tools/intellij/stackdriver/debugger/StackdriverDebuggerIcons.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.stackdriver.debugger;
+
+import com.intellij.openapi.util.IconLoader;
+import javax.swing.Icon;
+import javax.swing.ImageIcon;
+
+public class StackdriverDebuggerIcons {
```
Comment (etanshaul) [inline google-cloud-tools-plugin/stackdriver-debugger/src/com/google/cloud/tools/intellij/stackdriver/debugger/StackdriverDebuggerIcons.java]: annoying nit, but javadoc?

File: google-cloud-tools-plugin/stackdriver-debugger/src/com/google/cloud/tools/intellij/stackdriver/debugger/StackdriverDebuggerBundle.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+ */
+
+package com.google.cloud.tools.intellij.stackdriver.debugger;
+
+import com.intellij.CommonBundle;
+import java.util.ResourceBundle;
+import org.jetbrains.annotations.NonNls;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.PropertyKey;
+
```
Comment (etanshaul) [inline google-cloud-tools-plugin/stackdriver-debugger/src/com/google/cloud/tools/intellij/stackdriver/debugger/StackdriverDebuggerBundle.java]: same here with the javadoc

File: google-cloud-tools-plugin/ultimate/build.gradle [hunk 4]
```diff
@@ -36,6 +36,7 @@ compileJava.dependsOn project(':google-cloud-tools-plugin').tasks.getByName("com
 dependencies {
     compile files(project(':google-cloud-tools-plugin').sourceSets.main.output)
     compile project(':google-cloud-tools-plugin:google-cloud-core')
+    compile project(':google-cloud-tools-plugin:stackdriver-debugger')
```
Comment (etanshaul) [inline google-cloud-tools-plugin/ultimate/build.gradle]: is this dependency necessary (Ultimate -> debugger)?

File: google-cloud-tools-plugin/testSrc/com/google/cloud/tools/intellij/PropertiesTest.java [hunk 5]
```diff
@@ -35,9 +35,7 @@ public void testSpace() throws IOException {
       in = new FileInputStream(f);
       Properties p = new Properties();
       p.load(new InputStreamReader(in, "UTF-8"));
-      Assert.assertEquals(
-          "Do you want to checkout branch {0} and restore the saved stash?",
-          p.getProperty("clouddebug.restorestash"));
+      Assert.assertEquals("Google Cloud Tools", p.getProperty("cloud.tools.group.menu.text"));
```
Comment (ivanporty) [inline google-cloud-tools-plugin/testSrc/com/google/cloud/tools/intellij/PropertiesTest.java]: This test looks pretty useless in general, how about I'll file a separate PR to remove it?

File: google-cloud-tools-plugin/resources/messages/CloudToolsBundle.properties (hunk 1 above)
Comment (patflynn) [reply google-cloud-tools-plugin/resources/messages/CloudToolsBundle.properties]: no.   It's used by the the login extender which stays in that module

File: google-cloud-tools-plugin/ultimate/build.gradle (hunk 4 above)
Comment (patflynn) [reply google-cloud-tools-plugin/ultimate/build.gradle]: yep.  AppEngineServerInstance and others reference the debugger.

Comment (patflynn) [reply google-cloud-tools-plugin/ultimate/build.gradle]: JavaeeFrameworkSupportProviderTestCase

Comment (patflynn) [reply google-cloud-tools-plugin/ultimate/build.gradle]: let me see if I can move it to the test deps

File: google-cloud-tools-plugin/testSrc/com/google/cloud/tools/intellij/PropertiesTest.java (hunk 5 above)
Comment (patflynn) [reply google-cloud-tools-plugin/testSrc/com/google/cloud/tools/intellij/PropertiesTest.java]: knock yourself out. :)

File: google-cloud-tools-plugin/ultimate/build.gradle (hunk 4 above)
Comment (patflynn) [reply google-cloud-tools-plugin/ultimate/build.gradle]: moved the dep to testCompile

**Pull request merged**